pkg/ruler/frontend_decoder.go

```diff
@@ -19,13 +21,8 @@ type JsonDecoder struct{}
 type Warning []string
 
 func (j JsonDecoder) Decode(body []byte) (promql.Vector, Warning, error) {
-	var response struct {
-		Status    string          `json:"status"`
-		Data      json.RawMessage `json:"data"`
-		ErrorType string          `json:"errorType"`
-		Warnings  Warning         `json:"warnings"`
-		Error     string          `json:"error"`
-	}
+	var response api.Response
```

Comment:
We still need to add `infos` field to `api.Response` struct

Reply:
I have added it.